demo/useCryptKeeper.ts

```diff
@@ -0,0 +1,210 @@
+    const externalNullifier = encodeBytes32String("voting-1");
+    const signal = encodeBytes32String("hello-world");
+    const rlnIdentifier = RLN._genIdentifier();
+    const rlnIdentifierHex = bigintToHex(rlnIdentifier);
+
+    let storageAddressOrArtifacts: any = `${merkleStorageAddress}/RLN`;
+
+    if (proofType === MerkleProofType.ARTIFACTS) {
+      if (!mockIdentityCommitments.includes(selectedIdentity.commitment)) {
+        mockIdentityCommitments.push(selectedIdentity.commitment);
+      }
+
+      storageAddressOrArtifacts = {
+        leaves: mockIdentityCommitments,
+        depth: 15,
+        leavesPerNode: 2,
+      };
+    }
+
+    try {
+      const toastId = toast("Generating RLN proof...", {
+        type: "info",
+        hideProgressBar: true,
+        closeOnClick: true,
+        pauseOnHover: false,
+      });
+
+      const proof = await client?.rlnProof(externalNullifier, signal, storageAddressOrArtifacts, rlnIdentifierHex);
+
+      console.log("RLN proof generated successfully!", proof);
+      toast("RLN proof generated successfully!", { type: "success" });
+      toast.dismiss(toastId);
+    } catch (e) {
+      toast("Error while generating RLN proof!", { type: "error" });
+      console.error(e);
+    }
+  };
+
+  const getIdentityCommitment = useCallback(async () => {
+    const payload = await client?.getActiveIdentity();
+
+    if (!payload) {
+      return;
+    }
+
+    setSelectedIdentity({
+      commitment: payload.commitment,
+      web2Provider: payload.web2Provider,
+    });
+
+    toast(`Getting Identity Commitment successfully! ${payload.commitment}`, { type: "success" });
+  }, [client, setSelectedIdentity]);
+
+  const createIdentity = useCallback(() => {
+    client?.createIdentity();
+  }, [client]);
+
+  const onIdentityChanged = useCallback(
+    (payload: unknown) => {
+      const { commitment, web2Provider } = payload as SelectedIdentity;
+
+      setSelectedIdentity({ commitment, web2Provider });
+      toast(`Identity has changed! ${commitment}`, { type: "success" });
+    },
+    [setSelectedIdentity],
+  );
+
+  const onLogin = useCallback(() => {
+    setIsLocked(false);
+    getIdentityCommitment();
+  }, [setIsLocked, getIdentityCommitment]);
+
+  const onLogout = useCallback(() => {
+    setSelectedIdentity({
+      commitment: "",
+      web2Provider: "",
+    });
+    setIsLocked(true);
+  }, [setSelectedIdentity, setIsLocked]);
+
+  // Connect to CK
+  useEffect(() => {
+    const { cryptkeeper } = window;
+
+    if (!cryptkeeper) {
+      toast(`CryptKeeper is not installed in the browser`, { type: "error" });
+      return undefined;
+    }
+
+    if (!client) {
+      setClient(cryptkeeper);
+      return undefined;
+    }
+
+    client?.on("login", onLogin);
+    client?.on("identityChanged", onIdentityChanged);
+    client?.on("logout", onLogout);
+  }, [Boolean(client), onLogout, onIdentityChanged, onLogin, setClient, setIsLocked]);
```

Comment:
This hook misses the cleanup.

Reply:
It's needed to support this API: chrome.scripting.registerContentScripts